src/Agent.Plugins/TFCliManager.cs

```diff
@@ -275,7 +275,17 @@ public async Task WorkspaceDeleteAsync(ITfsVCWorkspace workspace)
 
         public async Task WorkspaceNewAsync()
         {
-            await RunCommandAsync("vc", "workspace", "/new", "/location:local", "/permission:Public", WorkspaceName);
+            var useServerWorkspace = StringUtil.ConvertToBoolean(ExecutionContext.Variables.GetValueOrDefault("build.useserverworkspaces")?.Value ?? "false");
+            ExecutionContext.Debug($"useServerWorkspace is set to : '{useServerWorkspace}'");
+
+            if (useServerWorkspace)
+            {
+                await RunCommandAsync("vc", "workspace", "/new", "/location:server", "/permission:Public", WorkspaceName);
```

Comment:
I understand the below code.  A shorter alternative would be what I have below.  It's shorter and makes it clear we're only flipping one flag.

`                await RunCommandAsync("vc", "workspace", "/new", useServerWorkspace == true ? "/location:server" : "/location:local", "/permission:Public", WorkspaceName); 
`